lib/saml/responses/base.rb

```diff
@@ -62,7 +62,8 @@ def map_message_to_error(error_message)
       end
 
       def authn_context_text
-        REXML::XPath.first(decrypted_document, '//saml:AuthnContextClassRef')&.text
+        response_doc = assertion_encrypted? ? decrypted_document : document
```

Comment:
This is the main change—`authn_context` below is left alone because it is used broadly for statsD metrics and several other methods. The implicit assumption is that assertions received by VA.gov will always been encrypted (there is PII in the assertions)